PYTHON_CDM_Conversion_code/daily_to_cdm_header_v1.py

```diff
@@ -1,42 +1,118 @@
-    
-###produce headre table using some of obs table column information 
-    hdf = pd.DataFrame() 
+    station : `str` 
+        Single station ID to process
+
+    subset : `str`
+        Path to file containing subset of IDs to process
 
-    ##need to separtate obervation id by "-" to extract station id    
-    hdf[['primary_station_id', 'station_record_number', '1',"2,","3"]] = merged_df['report_id'].str.split('-', expand=True)                                                    
-    #hdf["observation_id"]=merged_df["observation_id"]                                                  
+    run_all : `bool`
+        Run all files in the directory defined in the config file
+
+    clobber : `bool`
+        Overwrite existing files if they exist.  If False, will skip existing ones
+    """
+    # Read in either single file, list of files or run all
+
+    # Check for sensible inputs
+    if station != "" and subset != "" and all:
+        print("Please select either single station, list of stations run or to run all")
+        return
+    elif station == "" and subset == "" and not all:
+        print("Please select either single station, list of stations run or to run all")
+        return
+
+    # Obtain list of station(s) to process (single/subset/all)
+    if station != "":
+        print(f"Single station run: {station}")
+        all_filenames = [os.path.join(utils.DAILY_HEAD_IN_DIR, f"{station}.{EXTENSION}")]
+    elif subset != "":
+        print(f"Subset of stations run defined in: {subset}")
+        # Read filenames form a list of 5000 to parallel
+        try:
+            # e.g. "/work/scratch-pw/snoone/qff_cdm_test_2021/station_list/ls1.txt"
+            with open(subset, "r") as f:
+                filenames = f.read().splitlines()
+
+                # now add the path to the front
+                all_filenames = []
+                for infile in filenames:
+                    all_filenames += [os.path.join(utils.DAILY_HEAD_IN_DIR, f"{infile}")]
+
+            print(f"   N = {len(all_filenames)}")
+        except IOError:
+            print(f"Subset file {subset} cannot be found")
+            return
+        except OSError:
+            print(f"Subset file {subset} cannot be found")
+            return
+    elif all:
+        print(f"All stations run in {utils.DAILY_HEAD_IN_DIR}")
+        all_filenames = [i for i in glob.glob(os.path.join(utils.DAILY_HEAD_IN_DIR, f'*.{EXTENSION}'))]    
+        print(f"   N = {len(all_filenames)}")
+              
+    # To start at next file after last processes 
+    #for filename in all_filenames[all_filenames.index('FRI0000LFBF.qff'):] :
+
+    # To start at begining of files
+    for filename in all_filenames:
+        print(f"Processing {filename}") 
+        
+        col_list = ["observation_id", "report_id", "latitude","longitude","source_id","date_time"] 
+        merged_df=pd.read_csv(filename, sep="|", usecols=col_list)

```

Comment:
Python standards want a space before and after the `=` when assigning values (but not when passing things into subroutines).  Can you check this throughout? [here, line 99,...]